rtrlib/lib/vector.c

```diff
@@ -44,37 +47,125 @@ int delete_vector(struct lrtr_vector *vector)
 	return 0;
 }
 
-int reallocate_vector(struct lrtr_vector *vector)
+int reallocate_vector(struct aspa_vector *vector)
 {
 	// by how many more elements this vector can hold after the relocation
 	const size_t SIZE_INCREASE = 100;
 
 	// allocation the new chunk of memory
-	uint32_t *new_data_field = (uint32_t *)malloc(sizeof(uint32_t) * (vector->capacity + SIZE_INCREASE));
+	struct aspas_record *new_data_field =
+		(struct aspas_record *)malloc(sizeof(struct aspas_record) * (vector->capacity + SIZE_INCREASE));
 
 	// malloc failed so returning an error
 	if (new_data_field == NULL) {
 		return -1;
 	}
 
 	// copying the data from the old location to the new one
-	memcpy(new_data_field, vector->data, vector->capacity);
+	memcpy(new_data_field, vector->data, vector->capacity * sizeof(struct aspas_record));
+	vector->data = new_data_field;
 	vector->capacity += SIZE_INCREASE;
 
 	return 0;
 }
 
-int add_element_vector(struct lrtr_vector *vector, uint32_t value)
+// TODO: discuss if quicksort is the best for the use case or maybe an insertion sort is better
+// because it has linear complexity on insert
```

Comment:
agree, but would add insertion sort inline with insert_element function, which would assume the vector to be sorted already